src/app/modules/truenas-connect/components/truenas-connect-modal/truenas-connect-modal.component.ts

```diff
@@ -0,0 +1,143 @@
+import { WINDOW } from 'app/helpers/window.helper';
+import { helptextTopbar } from 'app/helptext/topbar';
+import { IxInputComponent } from 'app/modules/forms/ix-forms/components/ix-input/ix-input.component';
+import { IxIconComponent } from 'app/modules/ix-icon/ix-icon.component';
+import { AppLoaderService } from 'app/modules/loader/app-loader.service';
+import { TruenasConnectService } from 'app/modules/truenas-connect/services/truenas-connect.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-truenas-connect-modal',
+  standalone: true,
+  imports: [
+    MatDialogTitle,
+    MatDialogContent,
+    MatDialogActions,
+    IxIconComponent,
+    IxInputComponent,
+    MatButton,
+    ReactiveFormsModule,
+    TranslateModule,
+  ],
+  templateUrl: './truenas-connect-modal.component.html',
+  styleUrl: './truenas-connect-modal.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TruenasConnectModalComponent {
+  readonly helptext = helptextTopbar;
+  readonly TruenasConnectStatus = TruenasConnectStatus;
+  protected readonly requiredRoles = [Role.TrueCommandWrite];
+  protected form = this.formBuilder.group({
+    tnc_base_url: [this.tnc.config().tnc_base_url, [Validators.required]],
+    account_service_base_url: [this.tnc.config().account_service_base_url, [Validators.required]],
+    leca_service_base_url: [this.tnc.config().leca_service_base_url, [Validators.required]],
+  });
+
+  private formChanges = toSignal(this.form.valueChanges);
+  hasChange = computed(() => {
+    const initialValue = this.tnc.config();
+    const formValue = this.formChanges();
+    return Object.keys(formValue).some((key: keyof typeof formValue) => formValue[key] !== (initialValue)[key]);
+  });
+
+  protected readonly tooltips = {
+    account_service_base_url: T('Account Service Base Url'),
+    leca_service_base_url: T('Leca Service Base Url'),
+    tnc_base_url: T('TNC Base Url'),
+  };
+
+  constructor(
+    @Inject(WINDOW) private window: Window,
+    private loader: AppLoaderService,
+    private formBuilder: FormBuilder,
+    private dialogRef: MatDialogRef<TruenasConnectModalComponent>,
+    public tnc: TruenasConnectService,
+  ) {
+    effect(() => {
+      if (this.tnc.config().status === TruenasConnectStatus.Disabled) {
+        this.form.enable();
+      } else {
+        this.form.disable();
+      }
+    });
+  }
+
+  protected save(): void {
+    const formValue = this.form.getRawValue();
+    const payload = {
+      ips: this.tnc.config().ips,
+      enabled: true,
+      tnc_base_url: formValue.tnc_base_url,
+      account_service_base_url: formValue.account_service_base_url,
+      leca_service_base_url: formValue.leca_service_base_url,
+    };
+    this.loader.open();
+    this.tnc.enableService(payload)
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.loader.close();
+      });
+  }
+
+  protected enableService(): void {
```

Comment:
`save()` and `enableService` are similar, maybe you could call one from the other.